### DISCUSSION TITLE
API access via Personal Access Token broken since 5.6.3

### DISCUSSION THREAD

**Discussion by xpfgsyb** [firefly-iii/firefly-iii#5282] [Firefly III] [answered]
First the good news for JC5: it does work with your demo system, so I guess it's not a general bug and therefore I didn't open a ticket, but this discussion instead.

The problem is that since I updated Firefly to version 5.6.3 I'm not able to use API requests with a PAT like the following one:
`curl -X 'GET' 'https://my.firefly.url/api/v1/accounts?page=1' -H 'accept: application/vnd.api+json' -H 'Authorization: Bearer my-very-long-personal-access-token'`
Response from Firefly: `{"message":"Unauthenticated","exception":"AuthenticationException"}` and HTTP error 401.

I'm not sure what's the breaking factor. I'm using Firefly via composer install on a Debian system (11.1 with latest updates) and Debian 8.0 packages from sury.org. There were two changes between the working state a few days ago and the problem: I updated Firefly from 5.6.2 to 5.6.3 (and later 5.6.4, which doesn't work too) and I updated several PHP packages (bcmath, common, curl, gd, intl, json, ldap, mbstring, mysql, sqlite3, xml, zip) from `2:8.0+85+0~20211104.37+debian11~1.gbp73ad3b` to `2:8.0+85+0~20211110.38+debian11~1.gbp73ad3b`.

I was already able to confirm that it also doesn't work with the older PHP packages and Firefly 5.6.4, so I guess the PHP update wasn't the reason why it broke. Where there any changes to the Firefly authentication via PAT? Or is maybe the PHP version on your demo server a bit older than mine and the breaking change were in PHP between these versions? Does anyone has the same problem maybe or an idea what I can try?

**Comment by JC5** [comment, marked as answer]
Just a quick reply, were you able to confirm changing / replacing the PAT keys doesn't work either?

The code itself hasn't been changed in ages.

**Reply by xpfgsyb** [reply]
Wow, yes, a new PAT key did the trick. Thank you for the quick response!
Do they fail from time to time or had I just bad luck?

**Reply by JC5** [reply]
It could be your local keypair was gone or broken, PAT's should be valid forever.
